layouts/posts/list.html

```diff
@@ -3,7 +3,13 @@
 {{ end }}
 {{ define "content" }}
   <section class="container list">
-    <h1 class="title">{{ .Title }}</h1>
+    <h1 class="title">
+      {{- if i18n .Title}}
```

Comment:
I think is better to do something like this `{{ title (i18n (lower .Title)) | default .Title }}`. So, you should put "posts" as id in the language file. Yep, lower case.
Also, you need to translate the title inside the other `define`. This is the title shown in the browser tab/topbar.